# Elementary/Middle T1 21/22 Old-New Similarity Check

In [1]:
import pandas as pd
import os

In [2]:
old = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\rt_fisher_data.xlsx", sheet_name="TK-5")

In [3]:
old.head()

School Grade Level  Student ID Gender  \
0  Berkeley Arts Magnet Elementary School           K     2020428      F   
1  Berkeley Arts Magnet Elementary School           K     2020105      F   
2  Berkeley Arts Magnet Elementary School           K     2020091      F   
3  Berkeley Arts Magnet Elementary School           K     2020130      M   
4  Berkeley Arts Magnet Elementary School           K     2020653      M   

  Reported Race Student Is Special Ed? 504 English Proficiency Language  \
0         White                      N   N                  EO      NaN   
1        Latinx                      N   N                  EO      NaN   
2         White                      N   N                  EO      NaN   
3         White                      N   N                  EO      NaN   
4         White                      N   N                  EO      NaN   

  SED from CALPADS  ... STAR Reading Scaled Score  STAR Math Spanish?  \
0                N  ...                       NaN                 NaN   
1                N  ...                       NaN                 NaN   
2                N  ...                       NaN                 NaN   
3                N  ...                       NaN                 NaN   
4                N  ...                       NaN                 NaN   

   STAR Math District BC Name STAR Math Percentile Rank  \
0                         NaN                       NaN   
1                         NaN                       NaN   
2                         NaN                       NaN   
3                         NaN                       NaN   
4                         NaN                       NaN   

   STAR Math Scaled Score IDEL Instructional Level  Dibels Composite Score  \
0                     NaN                      NaN                   327.0   
1                     NaN                      NaN                   275.0   
2                     NaN                      NaN                   302.0   
3                     NaN                      NaN                   328.0   
4                     NaN                      NaN                   332.0   

   Dibels Composite Level  Math Benchmark Percent Correct  \
0            At Benchmark                           86.67   
1    Well Below Benchmark                              40   
2         Below Benchmark                           86.67   
3            At Benchmark                             100   
4         Above Benchmark                           86.67   

   Math Benchmark Performance Level  
0                      Consistently  
1                            Rarely  
2                      Consistently  
3                      Consistently  
4                      Consistently  

[5 rows x 34 columns]

In [4]:
old_cols = old.columns.to_list()

In [5]:
old_cols

['School',
 'Grade Level',
 'Student ID',
 'Gender',
 'Reported Race',
 'Student Is Special Ed?',
 504,
 'English Proficiency',
 'Language',
 'SED from CALPADS',
 'McKV/ Homeless',
 'Suspensions through 10/25',
 'Attendance through 10/15',
 'Report Card: English TCRWP',
 'Report Card: English TCRWP level',
 'Report Card: Spanish TCRWP',
 'Report Card: Spanish TCRWP level',
 'Report Card: Op & Alg Thinking',
 'Report Card: Num & Ops in Base Ten',
 'Report Card: Measurement and Data',
 'Report Card: Geometry',
 'Report Card: Math Fluency',
 'STAR Reading District BC Name',
 'STAR Reading Instructional Reading Level',
 'STAR Reading Scaled Score',
 'STAR Math Spanish?',
 'STAR Math District BC Name',
 'STAR Math Percentile Rank',
 'STAR Math Scaled Score',
 'IDEL Instructional Level',
 'Dibels Composite Score',
 'Dibels Composite Level',
 'Math Benchmark Percent Correct',
 'Math Benchmark Performance Level']

In [6]:
new = old = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\Elementary_Middle T1 21_22.xlsx", sheet_name="Sheet1")

In [7]:
new.head()

School Grade Level         Teacher  \
0  Berkeley Arts Magnet Elementary School           K  Stroik, Sophia   
1  Berkeley Arts Magnet Elementary School           K  Stroik, Sophia   
2  Berkeley Arts Magnet Elementary School           K  Stroik, Sophia   
3  Berkeley Arts Magnet Elementary School           K  Stroik, Sophia   
4  Berkeley Arts Magnet Elementary School           K  Stroik, Sophia   

   Student ID  Last Name First Name Gender Reported Race  \
0   2020428.0    Adelman     Margot      F         White   
1   2020105.0   Andrizzi    Liliana      F        Latinx   
2   2020091.0     Arnold        Aya      F         White   
3   2020130.0     Benton    Charles      M         White   
4   2020653.0  Berkowitz       Levi      M         White   

  Student Is Special Ed? 504.0  ... STAR Reading Scaled Score  \
0                      N     N  ...                       NaN   
1                      N     N  ...                       NaN   
2                      N     N  ...                       NaN   
3                      N     N  ...                       NaN   
4                      N     N  ...                       NaN   

  STAR Math Spanish? STAR Math District BC Name STAR Math Percentile Rank  \
0                NaN                        NaN                       NaN   
1                NaN                        NaN                       NaN   
2                NaN                        NaN                       NaN   
3                NaN                        NaN                       NaN   
4                NaN                        NaN                       NaN   

   STAR Math Scaled Score  IDEL Instructional Level Dibels Composite Score  \
0                     NaN                       NaN                  327.0   
1                     NaN                       NaN                  275.0   
2                     NaN                       NaN                  302.0   
3                     NaN                       NaN                  328.0   
4                     NaN                       NaN                  332.0   

   Dibels Composite Level Math Benchmark Percent Correct  \
0            At Benchmark                          86.67   
1    Well Below Benchmark                             40   
2         Below Benchmark                          86.67   
3            At Benchmark                            100   
4         Above Benchmark                          86.67   

   Math Benchmark Performance Level  
0                      Consistently  
1                            Rarely  
2                      Consistently  
3                      Consistently  
4                      Consistently  

[5 rows x 37 columns]

In [8]:
new_cols = new.columns.to_list()

In [9]:
new_cols

['School',
 'Grade Level',
 'Teacher',
 'Student ID',
 'Last Name',
 'First Name',
 'Gender',
 'Reported Race',
 'Student Is Special Ed?',
 504.0,
 'English Proficiency',
 'Language',
 'SED from CALPADS',
 'McKV/ Homeless',
 'Suspensions through 10/25',
 'Attendance through 10/15',
 'Report Card: English TCRWP',
 'Report Card: English TCRWP level',
 'Report Card: Spanish TCRWP',
 'Report Card: Spanish TCRWP level',
 'Report Card: Op & Alg Thinking',
 'Report Card: Num & Ops in Base Ten',
 'Report Card: Measurement and Data',
 'Report Card: Geometry',
 'Report Card: Math Fluency',
 'STAR Reading District BC Name',
 'STAR Reading Instructional Reading Level',
 'STAR Reading Scaled Score',
 'STAR Math Spanish?',
 'STAR Math District BC Name',
 'STAR Math Percentile Rank',
 'STAR Math Scaled Score',
 'IDEL Instructional Level',
 'Dibels Composite Score',
 'Dibels Composite Level',
 'Math Benchmark Percent Correct',
 'Math Benchmark Performance Level']

In [10]:
len(old_cols)

34

In [11]:
len(new_cols)

37

In [12]:
i=0
for index, col in enumerate(old_cols):
    if col in new_cols:
        i+=1
        #print(f"col {index} of old data is in the new data")

In [13]:
i

34

In [14]:
i=0
for index, col in enumerate(old_cols):
    if col in new_cols:
        i+=1
        print(f"col {index} of old data is in the new data")

col 0 of old data is in the new data
col 1 of old data is in the new data
col 2 of old data is in the new data
col 3 of old data is in the new data
col 4 of old data is in the new data
col 5 of old data is in the new data
col 6 of old data is in the new data
col 7 of old data is in the new data
col 8 of old data is in the new data
col 9 of old data is in the new data
col 10 of old data is in the new data
col 11 of old data is in the new data
col 12 of old data is in the new data
col 13 of old data is in the new data
col 14 of old data is in the new data
col 15 of old data is in the new data
col 16 of old data is in the new data
col 17 of old data is in the new data
col 18 of old data is in the new data
col 19 of old data is in the new data
col 20 of old data is in the new data
col 21 of old data is in the new data
col 22 of old data is in the new data
col 23 of old data is in the new data
col 24 of old data is in the new data
col 25 of old data is in the new data
col 26 of old data is 

## T2 21-22 (Elementary)

In [15]:
new_t2 = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\21_22 Data Sheets TK-12 RT.xlsx", sheet_name="Elementary T2")

In [16]:
new_t2.head()

School Grade Level             Teacher  \
0  Thousand Oaks Elementary School         5.0    Gillis, Jennifer   
1      John Muir Elementary School         5.0     Rosenthal, Lisa   
2       Cragmont Elementary School          KG        Rosen, Adria   
3       Cragmont Elementary School         2.0  Herbertson, Janine   
4      John Muir Elementary School         5.0     Rosenthal, Lisa   

   Student ID       Last Name First Name Gender Reported Race  \
0   2013206.0           Abbas   Muhammad      M         Asian   
1   2010668.0  Abcarian-Scott    Anahita      F        Latinx   
2   2020100.0        Abdallah     Shatha      F         White   
3   2016531.0        Abdallah     Shayma      F         White   
4   2018595.0    Abdel Rahman      Asser      M         White   

  Student Is Special Ed?     504.0  ... STAR Math Spanish?  \
0                    NaN       NaN  ...                NaN   
1                      Y  504 Plan  ...                NaN   
2                    NaN       NaN  ...                NaN   
3                    NaN       NaN  ...                NaN   
4                    NaN       NaN  ...                NaN   

  Spanish Math  District BCL Spanish Math District Benchmark Name  \
0                        NaN                                  NaN   
1                        NaN                                  NaN   
2                        NaN                                  NaN   
3                        NaN                                  NaN   
4                        NaN                                  NaN   

  Spanish Math Percentile Rank Spanish Math Scaled Score  \
0                          NaN                       NaN   
1                          NaN                       NaN   
2                          NaN                       NaN   
3                          NaN                       NaN   
4                          NaN                       NaN   

   IDEL Instructional Level  Dibels Composite Score Dibels Composite Level  \
0                       NaN                     NaN                    NaN   
1                       NaN                     NaN                    NaN   
2                       NaN                   341.0   Well Below Benchmark   
3                       NaN                   380.0        Below Benchmark   
4                       NaN                     NaN                    NaN   

  Math Benchmark Percent Correct Math Benchmark Performance Level  
0                            NaN                              NaN  
1                            NaN                              NaN  
2                            NaN                              NaN  
3                         0.4872      Needing Significant Support  
4                            NaN                              NaN  

[5 rows x 45 columns]

In [17]:
new_t2_cols_=new_t2.columns.to_list()

In [18]:
new_t2_cols=[]
for col in new_t2_cols_:
    string_col = str(col)
    new_t2_cols.append(string_col)

In [19]:
new_t2_cols

['School',
 'Grade Level',
 'Teacher',
 'Student ID',
 'Last Name',
 'First Name',
 'Gender',
 'Reported Race',
 'Student Is Special Ed?',
 '504.0',
 'English Proficiency',
 'Language',
 'SED from CALPADS',
 'McKV/ Homeless',
 'Vision Scholars',
 'Suspensions through 3/4/22',
 'Attendance through 3/4/22',
 'Writing On Demand',
 'Report Card: English TCRWP',
 'Report Card: English TCRWP level',
 'Report Card: Spanish TCRWP',
 'Report Card: Spanish TCRWP level',
 'Report Card: Op & Alg Thinking',
 'Report Card: Num & Ops in Base Ten',
 'Report Card: Measurement and Data',
 'Report Card: Geometry',
 'Report Card: Math Fluency',
 'STAR Reading District BC Name',
 'STAR Reading State BC Name ',
 'STAR Reading Instructional Reading Level',
 'STAR Reading Scaled Score',
 'STAR Math District BC Name',
 'STAR Math State BC Name',
 'STAR Math Percentile Rank',
 'STAR Math Scaled Score',
 'STAR Math Spanish?',
 'Spanish Math  District BCL',
 'Spanish Math District Benchmark Name',
 'Spanish Math 

In [20]:
i=0
for index, col in enumerate(old_cols):
    if col in new_t2_cols:
        i+=1
        #print(f"col {index} of old data is in the new data")
    else:
        print(f"{col} col isnt in T2 elementary new data")

print(i)


504 col isnt in T2 elementary new data
Suspensions through 10/25 col isnt in T2 elementary new data
Attendance through 10/15 col isnt in T2 elementary new data
31


In [21]:
i=0
for index, col in enumerate(old_cols):
    if col in new_t2_cols:
        i+=1
        print(f"col {index} of old data is in the new data")

col 0 of old data is in the new data
col 1 of old data is in the new data
col 2 of old data is in the new data
col 3 of old data is in the new data
col 4 of old data is in the new data
col 5 of old data is in the new data
col 7 of old data is in the new data
col 8 of old data is in the new data
col 9 of old data is in the new data
col 10 of old data is in the new data
col 13 of old data is in the new data
col 14 of old data is in the new data
col 15 of old data is in the new data
col 16 of old data is in the new data
col 17 of old data is in the new data
col 18 of old data is in the new data
col 19 of old data is in the new data
col 20 of old data is in the new data
col 21 of old data is in the new data
col 22 of old data is in the new data
col 23 of old data is in the new data
col 24 of old data is in the new data
col 25 of old data is in the new data
col 26 of old data is in the new data
col 27 of old data is in the new data
col 28 of old data is in the new data
col 29 of old data is

### Findings

**1. T1 21-22 (elementary) has all 34/34 of old cols. View output of next cell to delete the ones (3 exactly) that are extra**

**2. T2 21-22 (elementary) has 31/34 of old cols. Col 504.0 and 504 requires renaming (32/34). Suspensions through 10/25 and Attendance through 10/25 cols are misssing (probe further)**

In [22]:
for index, col in enumerate(new_cols):
    if col not in new_t2_cols:
        print(f"{col} col from new elementary data (T1 21-22) must be deleted")

504.0 col from new elementary data (T1 21-22) must be deleted
Suspensions through 10/25 col from new elementary data (T1 21-22) must be deleted
Attendance through 10/15 col from new elementary data (T1 21-22) must be deleted


# 21/22 Data Sheets TK-12 RT Old-New Similarity Check

In [23]:
new_king = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\21_22 Data Sheets TK-12 RT.xlsx", sheet_name="King S1")

In [24]:
new_king.head()

School  Grade Level  Student ID     Last Name  \
0  Martin Luther King Middle School          6.0   2021427.0  Abbate-Kahan   
1  Martin Luther King Middle School          6.0   2021427.0  Abbate-Kahan   
2  Martin Luther King Middle School          6.0   2021427.0  Abbate-Kahan   
3  Martin Luther King Middle School          6.0   2021427.0  Abbate-Kahan   
4  Martin Luther King Middle School          6.0   2021427.0  Abbate-Kahan   

   First Name Period     Course          Teacher S1 Grade Grading Task   ...  \
0  Tomas Leon    1+2    ELL 1-2  Taylor, Marcela      4.0           HOW  ...   
1  Tomas Leon    1+2    ELL 1-2  Taylor, Marcela      4.0           SBP  ...   
2  Tomas Leon    3.0     Math 6     Keeley, Ryan      4.0           HOW  ...   
3  Tomas Leon    3.0     Math 6     Keeley, Ryan      4.0           SBP  ...   
4  Tomas Leon    4.0  Science 6     Keeley, Ryan      3.0           HOW  ...   

  Semester One Suspensions  Reported Race Special Ed EL Status Language  \
0                       NaN      Hispanic          N   Not LEP      NaN   
1                       NaN      Hispanic          N   Not LEP      NaN   
2                       NaN      Hispanic          N   Not LEP      NaN   
3                       NaN      Hispanic          N   Not LEP      NaN   
4                       NaN      Hispanic          N   Not LEP      NaN   

   Gender  Ind Study?  Socio-Ec Disadv. 504? McKV/Homeless  
0       M         NaN                 N  NaN           NaN  
1       M         NaN                 N  NaN           NaN  
2       M         NaN                 N  NaN           NaN  
3       M         NaN                 N  NaN           NaN  
4       M         NaN                 N  NaN           NaN  

[5 rows x 27 columns]

In [25]:
new_king_cols = new_king.columns.to_list()

In [26]:
new_king_cols

['School',
 'Grade Level',
 'Student ID',
 'Last Name',
 'First Name',
 'Period',
 'Course',
 'Teacher',
 'S1 Grade',
 'Grading Task ',
 'Winter STAR Reading District Benchmark Cat.',
 'Winter STAR Reading State Benchmark Cat.',
 'Winter STAR Reading Inst. Reading Level',
 'Winter STAR Math District Benchmark Cat.',
 'Winter STAR Math State Benchmark Cat.',
 'Winter STAR Math Percentile Rank',
 'Semester One Attendance ',
 'Semester One Suspensions ',
 'Reported Race',
 'Special Ed',
 'EL Status',
 'Language',
 'Gender',
 'Ind Study?',
 'Socio-Ec Disadv.',
 '504?',
 'McKV/Homeless']

In [27]:
old_king = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\rt_fisher_data.xlsx", sheet_name="Q1 Grades")

In [28]:
old_king.head()

School  Grade Level  Student ID Period           Course  \
0  Berkeley High School            9     2006000      1    Physics 1 (P)   
1  Berkeley High School            9     2006000      2          INE-CLC   
2  Berkeley High School            9     2006000      3  FrshSemHist (P)   
3  Berkeley High School            9     2006000      4        Spanish I   
4  Berkeley High School            9     2006000      5    English 1 (P)   

  Q1 Grade   Grading Task Fall STAR Reading Benchmark Cat.  \
0       B-  Quarter Grade                              NaN   
1        A  Quarter Grade                              NaN   
2       B-  Quarter Grade                              NaN   
3       B-  Quarter Grade                              NaN   
4       D-  Quarter Grade                              NaN   

  Fall STAR Reading Inst. Reading Level Fall STAR Math Benchmark Cat.  ...  \
0                                   NaN                           NaN  ...   
1                                   NaN                           NaN  ...   
2                                   NaN                           NaN  ...   
3                                   NaN                           NaN  ...   
4                                   NaN                           NaN  ...   

  Attendance 8/16 to 10/16  Suspensions through 10/15  Reported Race  \
0                   0.9767                        NaN          Asian   
1                   0.9767                        NaN          Asian   
2                   0.9767                        NaN          Asian   
3                   0.9767                        NaN          Asian   
4                   0.9767                        NaN          Asian   

  Special ED  ELA Status Gender Ind Study Socio-Ec Disadv. 504 McKV/Homeless  
0          Y  Exited LEP      F         N                Y   N             N  
1          Y  Exited LEP      F         N                Y   N             N  
2          Y  Exited LEP      F         N                Y   N             N  
3          Y  Exited LEP      F         N                Y   N             N  
4          Y  Exited LEP      F         N                Y   N             N  

[5 rows x 21 columns]

In [29]:
old_king_cols= old_king.columns.to_list()

In [30]:
old_king_cols
len(old_king_cols)

21

In [31]:
i=0
for index, col in enumerate(old_king_cols):
    if col in new_king_cols:
        i+=1
        #print(f"col {index} of old data is in the new data")
    else:
        print(f"{col} col not found")

print(i)


Q1 Grade col not found
Grading Task col not found
Fall STAR Reading Benchmark Cat. col not found
Fall STAR Reading Inst. Reading Level col not found
Fall STAR Math Benchmark Cat. col not found
Fall STAR Math Percentile Rank col not found
Attendance 8/16 to 10/16 col not found
Suspensions through 10/15 col not found
Special ED col not found
ELA Status col not found
Ind Study col not found
504 col not found
9


**For King S1, 09/21 found only. 11/21 missing or with other names**

In [32]:
new_longfellow = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\21_22 Data Sheets TK-12 RT.xlsx", sheet_name="Longfellow S1")

new_longfellow_cols = new_longfellow.columns.to_list()

old_longfellow = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\rt_fisher_data.xlsx", sheet_name="Q1 Grades")

old_longfellow_cols= old_longfellow.columns.to_list()

i=0
for index, col in enumerate(old_longfellow_cols):
    if col in new_longfellow_cols:
        i+=1
        #print(f"col {index} of old data is in the new data")
    else:
        print(f"{col} col not found")

print(i)


School col not found
Q1 Grade col not found
Grading Task col not found
Fall STAR Reading Benchmark Cat. col not found
Fall STAR Reading Inst. Reading Level col not found
Fall STAR Math Benchmark Cat. col not found
Fall STAR Math Percentile Rank col not found
Attendance 8/16 to 10/16 col not found
Suspensions through 10/15 col not found
Special ED col not found
ELA Status col not found
Ind Study col not found
504 col not found
8


**For Longfellow S1, 08/21 found only. 13/21 missing or with other names**

In [33]:
new_williard = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\21_22 Data Sheets TK-12 RT.xlsx", sheet_name="Willard S1")

new_williard_cols = new_williard.columns.to_list()

old_williard = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\rt_fisher_data.xlsx", sheet_name="Q1 Grades")

old_williard_cols= old_williard.columns.to_list()

i=0
for index, col in enumerate(old_williard_cols):
    if col in new_williard_cols:
        i+=1
        #print(f"col {index} of old data is in the new data")
    else:
        print(f"{col} col not found")

print(i)


Q1 Grade col not found
Grading Task col not found
Fall STAR Reading Benchmark Cat. col not found
Fall STAR Reading Inst. Reading Level col not found
Fall STAR Math Benchmark Cat. col not found
Fall STAR Math Percentile Rank col not found
Attendance 8/16 to 10/16 col not found
Suspensions through 10/15 col not found
Special ED col not found
ELA Status col not found
Ind Study col not found
504 col not found
9


**For Williard S1, 09/21 found only. 12/21 missing or with other names**

In [34]:
new_bhs = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\21_22 Data Sheets TK-12 RT.xlsx", sheet_name="BHSBTA S1")

new_bhs_cols = new_bhs.columns.to_list()

old_bhs = pd.read_excel(r"C:\Users\Hussam\Desktop\RT Fisher\rt_fisher_data.xlsx", sheet_name="Q1 Grades")

old_bhs_cols= old_bhs.columns.to_list()

i=0
for index, col in enumerate(old_bhs_cols):
    if col in new_bhs_cols:
        i+=1
        #print(f"col {index} of old data is in the new data")
    else:
        print(f"{col} col not found")

print(i)


Q1 Grade col not found
Grading Task col not found
Fall STAR Reading Benchmark Cat. col not found
Fall STAR Reading Inst. Reading Level col not found
Fall STAR Math Benchmark Cat. col not found
Fall STAR Math Percentile Rank col not found
Attendance 8/16 to 10/16 col not found
Suspensions through 10/15 col not found
Special ED col not found
ELA Status col not found
Ind Study col not found
Socio-Ec Disadv. col not found
504 col not found
8


**For BHS/BTA S1, 08/21 found only. 13/21 missing or with other names**